### PR TITLE
New keymap for the dztech/dz65rgb

### PR DESCRIPTION
**Pull request by catrielmuller**
## Description

- Custom keymap center on the Linux functionality
- The left shift dispatch a `KC_APP` when it's tapped to be configured as a Composed Key on Linux
- Layer Index
- Function Layer for Temporary Function Keys and Mouse
- Multimedia Control Layer
- Emoji Unicode Layer ❤
- RGB Configuration Layer
- General Config Layer

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    [UC_THUMBSUP] = 0x1F44D,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_MAIN] = LAYOUT_65_ansi(
+        KC_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC, KC_HOME, \
+        KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_END, \
+        KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_ENT, KC_PGUP, \
+        KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, RSFT_T(KC_APP), KC_UP, KC_PGDN, \
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_LEAD, TO(_INDEX), MO(_FN), KC_LEFT, KC_DOWN, KC_RGHT \
```
Comment (fauxpark) [inline keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

File: keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        KC_NO, UC_RMOD, UC_MOD, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_ACL0, \
+        KC_NO, MAGIC_TOGGLE_NKRO, MAGIC_UNHOST_NKRO, MAGIC_HOST_NKRO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, \
+        KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, TO(_INDEX), KC_NO, KC_NO, KC_NO, KC_NO \
+    ),
+};
+
+void rgb_matrix_indicators_user(void) {
+
+    // CapsLock Light
+    if (IS_LED_ON(host_keyboard_leds(), USB_LED_CAPS_LOCK)) {
```
Comment (fauxpark) [inline keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c]:
```suggestion
    if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK)) {
```

File: keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    // Init
+}
+
+void matrix_scan_user(void) {
+    // User matrix
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record){
+    return true;
+}
```
Comment (fauxpark) [inline keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c]: These functions can be removed, as they're not doing anything.

File: keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c (hunk 1 above)
Comment (catrielmuller) [reply keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c]: Changed on the commit (8e136ab)

File: keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c (hunk 2 above)
Comment (catrielmuller) [reply keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c]: Thanks for this one. I did not know this function.
Changed on the commit (8e136ab)

File: keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c (hunk 3 above)
Comment (catrielmuller) [reply keyboards/dztech/dz65rgb/keymaps/catrielmuller/keymap.c]: Changed on the commit (8e136ab)

**Pull request merged**